### PR TITLE
Updates Sphinx Docs

### PR DESCRIPTION
**Pull request by danehans**
- Adds markdown docs to Sphinx index.
- Lowercases docs names.

### CODE REVIEW

**Comment by danehans** [comment]
Marking as draft until other docs get merged. I'll then update the index and push a new commit.

File: docs/index.rst [hunk 1]
```diff
@@ -20,7 +20,9 @@ standalone or Kubernetes-based application gateway.
 
   intro/index
   intro/compatibility
-  user/QUICKSTART
```
Comment (Xunzhuo) [inline docs/index.rst:23]: This should be careful, if you only change QUCkSTART.md to quickstart.md, no changes will be made to git workspace

**Review by Xunzhuo — CHANGES_REQUESTED** [review]

File: docs/index.rst (hunk 1 above)
Comment (Xunzhuo) [reply docs/index.rst:23]: Sphinx will not find user/quickstart, so it will disappear..

File: docs/index.rst [hunk 2]
```diff
@@ -20,7 +20,9 @@ standalone or Kubernetes-based application gateway.
...
+  user/quickstart
+  user/http-routing
+  user/http-request-headers
   design/system-design
   design/ROADMAP
```
Comment (Xunzhuo) [inline docs/index.rst:27]: Here too, if you want everything in lowercase.

File: docs/index.rst (hunk 1 above)
Comment (danehans) [reply docs/index.rst:23]: I thought I updated all QUICKSTART.md references. Am I missing any?

Comment (Xunzhuo) [reply docs/index.rst:23]: Yes, you have updated all QUICKSTART.md referances, but the file name is still QUICKSTART.md not quickstart.md

Comment (Xunzhuo) [reply docs/index.rst:23]: So this ` user/quickstart` will be not found.

Comment (Xunzhuo) [reply docs/index.rst:23]: Or you should configure `git config core.ignorecase false` ?

**Comment by danehans** [comment]
https://github.com/envoyproxy/gateway/pull/595 added most of the changes in this PR. Add the missing http-request-headers.md to the index.

**Comment by danehans** [comment]
This PR is no longer needed since the changes were implemented in #614.

**Pull request closed without merging**